src/Dont/Exception/NonDeSerializableObject.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Dont\Exception;
+
+use LogicException;
+
+class NonDeSerializableObject extends LogicException implements ExceptionInterface
+{
+    public static function fromAttemptedDeSerialization($object) : self
+    {
+        if (! is_object($object)) {
+            throw TypeError::fromNonObject($object);
+        }
+
+        $template = <<<'ERROR'
```

Comment:
The idea of the lengthy message is to make this REALLY obnoxiously clear to the reader. There must be no issue opened by people asking why this kind of thing is being thrown (which is the reason why I started working on the package in first place)